src/cargo/util/config/mod.rs

```diff
@@ -1006,7 +1004,16 @@ impl Config {
             }
         }
 
-        cfg.merge(value, true)?;
+        if let CV::Table(map, _) = value {
```

Comment:
Can this maybe be simplified a little instead of removing and reinserting?  Maybe something like this?

```rust
        if let CV::Table(map, _) = value {
            let base_map = self.values_mut()?;
            for (k, v) in map {
                match base_map.entry(k) {
                    Vacant(entry) => { entry.insert(v); }
                    Occupied(mut entry) => { entry.get_mut().merge(v, true)? }
                }
            }
        }
```